### DISCUSSION TITLE
Reason behind choosing docker instead micro VMs (like firecracker)

### DISCUSSION THREAD

**Discussion by hmontazeri** [basecamp/kamal#94] [General]
Just curious as I read that fly.io uses firecracker micro VMs for their infrastructure. I was wondering the reasoning behind the decision to use docker as the app runner, which has a significant overhead compared to micro VMs. 

Projects like [ignite](https://github.com/weaveworks/ignite) make it easy to run docker images as micro VMs 😄 

Just wondering...

**Comment by haikyuu** [comment]
Ignite is quite interesting, it states it has a container like DX do `ignite run` is like `docker run`.

I wonder if it's possible to support ignite in mrsk. And how big a refactoring would that be

Firecracker VMs are super fast and lightweight, and it's even freebsd is [actively working on supporting firecracker](https://www.daemonology.net/blog/2022-10-18-FreeBSD-Firecracker.html) which means even software like PFsense will soon be able to run in firecracker.

**Comment by nickchomey** [comment]
I have to figure that docker was chosen because it was already chosen and use in their kubernetes cluster, and by everyone else. The goal here, it seems to me, was simply to get off cloud, not rearchitect the project. 

Though now that I look at ignite, it seems to make this a non-issue. So, it probably comes down to them just not being aware of these tools.

**Comment by christianbpedersen** [comment]
Wouldn't the performance gains here be dwarfed by Rails' appetite for resources? It seems a bit theoretical.
